File: big-data-analytics/MapReduce/src/main/java/GrepTheWeb/GrepTheWeb.java
package GrepTheWeb;

public class GrepTheWeb {

	public static void main(String[] args) throws Exception {
		if (args.length != 3) {
			System.out.println("Grep <inDir> <outDir> <regex>");
			System.exit(2);
		}
		String input = args[0];
		String output = args[1];
		String regex = args[2];
		ProcessBuilder pb = new ProcessBuilder("python", "GrepTheWeb" + ".py",
				input, output, regex).inheritIO();
		Process p = pb.start();
		p.waitFor();
		System.exit(0);

	}

}

File: big-data-analytics/MapReduce/GrepTheWeb.py
import os
import sys

args = sys.argv
if len(args) != 4:
    print("python GrepTheWeb.py <inputFile> <outputDirectory> <regex>")
    sys.exit(1)
os.system("/usr/local/hadoop/bin/hadoop jar \
     /usr/local/hadoop/share/hadoop/tools/lib/hadoop-streaming-*.jar \
    -D mapred.reduce.tasks=1 \
    -input " + args[1] +" \
    -mapper grepmapper.py \
    -reducer grepreducer.py \
    -file grepmapper.py \
    -output " + args[2] +" \
    -cmdenv SEARCH_STRING=" + args[3])
sys.exit(1)
